packages/client/src/clients/in-app-message/message-client.ts

```diff
@@ -0,0 +1,109 @@
+import Knock from "../../knock";
+import { NetworkStatus, isRequestInFlight } from "../../networkStatus";
+
+import { InAppChannelClient } from "./channel-client";
+import {
+  FetchInAppMessagesOptions,
+  InAppMessageClientOptions,
+  InAppMessagesQueryInfo,
+} from "./types";
+
+/**
+ * Manages realtime connection to in app messages service.
+ */
+export class InAppMessageClient {
+  private knock: Knock;
+
+  constructor(
+    readonly channelClient: InAppChannelClient,
+    readonly messageType: string,
+    readonly defaultOptions: InAppMessageClientOptions = {},
+  ) {
+    this.knock = channelClient.knock;
+  }
+
+  async fetch(options: FetchInAppMessagesOptions = {}) {
+    // TODO: Create better abstraction for reading from / writing to store
+    const queryParams = {
+      ...this.defaultOptions,
+      ...options,
+      // Unset options that should not be sent to the API
+      __loadingType: undefined,
+      __fetchSource: undefined,
+    };
+
+    const queryKey = `${this.messageType}-${JSON.stringify(queryParams)}`;
+    const queryState = this.channelClient.store.state.queries[queryKey] ?? {
+      items: [],
+      loading: false,
+      networkStatus: NetworkStatus.ready,
+    };
+    const networkStatus = queryState.networkStatus;
+
+    // If there's an existing request in flight, then do nothing
+    if (networkStatus && isRequestInFlight(networkStatus)) {
+      return;
```

Comment:
Is okay to return `undefined` here though?

Reply:
Yeah, I think it's fine to return nothing here. We do the same thing in the feeds client.

Reply:
Hm that feels kinda wrong to me, that this function can return nothing or something (including how we do in the feeds client). It feels like it should return a uniform value with a network status indicating an in-flight request. 

Don't want to block on this though, we can come back to this.

Reply:
Yeah, I'll leave a todo comment. I wouldn't be opposed to including the network status in the response (although you can grab that from the store for the given query), but I'd want to think through what the status should be